### ISSUE TITLE
Where is the language runtime?

### ISSUE THREAD

**Issue by lazzarello** [nasser/---#7]
Is the file qlib.peg the core of the language? Is there a reference for things like functions and arguments?

I'm confused because most of the code in this repository is a web application written in JavaScript and Ruby, not a programming language with Arabic syntax.

**Comment by nasser** [comment]
The repository is a web application written in JavaScript in Ruby that _implements_ a programming language called قلب that is programed using an Arabic syntax.

The internal implementation is largely in English because it is built on JavaScript, but you can see the mapping of Arabic function names to JavasScript implementations in [primitives.js](https://github.com/nasser/---/blob/master/public/qlb/primitives.js). Examples of whole scripts written in قلب can be found [in the examples folder](https://github.com/nasser/---/tree/master/public/lib/amthila). You can interact with the [interpreter online](http://repl.qlblang.org) to see what it all looks like in action.

qlb.peg is the grammar, used by peg.js to parse قلب source code. The runtime is largely implemented in [qlb.js](https://github.com/nasser/---/blob/master/public/qlb/qlb.js).

There is no reference for functions or arguments at the moment. I plan on holding workshops at [Eyebeam](http://eyebeam.org) in the coming months, though, so I should have some documentation up by then.

Hope that clears things up!

**Comment by lazzarello** [comment]
Awesome! That clears it up. Thanks for your work on this. I've been thinking about a non-english programming language since the day I learned that Java supports UTF-8 for everything, not just string types.

I've been in work situations (I'm a professional hacker) where I saw first hand the "I'll just learn English so I can write software" problem. You picked an pretty hard place to start! Right-to-left non-latin glyphs!

Thanks again. I'm in NYC so I'll stop by some by Eyebeam.
